azext_iot/tests/test_iot_central_int.py

```diff
@@ -204,5 +204,41 @@ def test_central_device_registration_info(self):
 
         # since time taken for provisioning to complete is not known
         # we can only assert that the payload is populated, not anything specific beyond that
-        assert json_result["central_info"] is not None
+        assert json_result["device_info"] is not None
         assert json_result["dps_state"] is not None
+
+    def test_central_device_registration_info_filter(self):
+        device_id = self.create_random_name(prefix="aztest", length=24)
+        device_name = self.create_random_name(prefix="aztest", length=24)
+        device_status_expected = "unassociated"
+
+        self.cmd(
+            "iot central app device create --app-id {} -d {} --device-name {}".format(
+                APP_ID, device_id, device_name
+            ),
+            checks=[
+                self.check("approved", True),
+                self.check("displayName", device_name),
+                self.check("id", device_id),
+                self.check("simulated", False),
+            ],
+        )
+
+        result = self.cmd(
+            "iot central app device registration-info --app-id {} --ds {}".format(
```

Comment:
Do we have the necessary foundation in place to exercise assertions against devices with other possible device status values?

Reply:
Unfortunately we don't, iot central creating a device doesn't actually provision it so its somewhat difficult to get the device into the other states programatically during an IT.

I'll look into this as a follow up and see if theres anything we can do

Reply:
One option is to create an app with existing devices in different states and  then adding validations. IT will take dependency on external environment. Will  explore other options as well.